### ISSUE TITLE
False positive: UMAC_UNCALLABLE_METHOD_OF_ANONYMOUS_CLASS

### ISSUE THREAD

**Issue by clo-vis** [spotbugs/spotbugs#2891]
```java
import java.util.stream.Stream;

class Test {
    public static void main(String[] args) {
        record Foo(int a) {
            int aPlus1() {
                return a + 1;
            }

            int aPlus2() {
                return a + 2;
            }
        }
        var foo = new Foo(1);
        Stream.of(foo).mapToInt(Foo::aPlus1).forEach(System.out::println);
        Stream.of(foo).mapToInt(r -> r.aPlus2()).forEach(System.out::println);
    }
}
```

Warning: _Uncallable method Test$1Foo.aPlus1() defined in anonymous class [Scary(5), High confidence]_

But:

1) There is no anonymous class (all classes in the example have a name)
2) Both methods aPlus2 and aPlus1 are callable and *are* called.

**Comment by clo-vis** [comment]
Sorry, just saw I already opened a similar bug 6 years ago.
